platform/iphone/camera_ios.mm

```diff
@@ -0,0 +1,441 @@
+/*************************************************************************/
+/*  camera_ios.mm                                                        */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                    http://www.godotengine.org                         */
+/*************************************************************************/
+/* Copyright (c) 2007-2017 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2017 Godot Engine contributors (cf. AUTHORS.md)    */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+///@TODO this is a near duplicate of CameraOSX, we should find a way to combine those to minimise code duplication!!!!
+// If you fix something here, make sure you fix it there as wel!
+
+#include "camera_ios.h"
+#include "servers/camera/camera_feed.h"
+
+#import <AVFoundation/AVFoundation.h>
+#import <UIKit/UIKit.h>
+
+//////////////////////////////////////////////////////////////////////////
+// MyCaptureSession - This is a little helper class so we can capture our frames
+
+@interface MyCaptureSession : AVCaptureSession <AVCaptureVideoDataOutputSampleBufferDelegate> {
```

Comment:
I'm not sure that we even need this class. The entire capture and feed process seems wholly unneccessary to me given that [ARKit provides](https://developer.apple.com/documentation/arkit/arframe/2867984-capturedimage?language=objc) a `CVPixelBufferRef` in their API. 

Surely you're duplicating the capture work, and it's being done in a slow method compared to other options. More details about [Apple's recommended approach](https://developer.apple.com/documentation/arkit/displaying_an_ar_experience_with_metal?language=objc#2891877) for home-grown ARKit.

Reply:
Also probably why you're not getting ambient light estimation.

Reply:
Thanks Sam! I'm going to look into that, I've been wondering about that for awhile now. Remember when I started ARKit that was when ARKIt 1.0 was just out so all this stuff wasn't available yet.

Reply:
Ah!! I see now. I am using the CVPixelBufferRef for ARKIt. The class you are looking at is part of the camera server logic that ARKit is build ontop of. See #10643.

Camera_ios is to obtain the camera image on iOS if you want to use that for whatever reason you have. It is not used by ARKit seeing as you state ARKit already does this for us, but you do need it if you want to use the camera logic outside of ARKit.